### PR TITLE
enhance: Remove "Hide countries with <1m people" checkbox

### LINKED ISSUE #1351: Remove "Hide countries with <1m people" checkbox
## Problem

> When there are no countries with a population less than 1 million, then there shouldn't be the option to 'hide countries < 1million people' shown. https://ourworldindata.org/grapher/q2-gdp-growth-vs-confirmed-deaths-due-to-covid-19-per-million-people

https://user-images.githubusercontent.com/1308115/160832713-7d65990c-8765-4328-9b5c-4c3319b1c96c.mp4

## Decision

We will completely remove the "hide countries < 1million people" checkbox, since it's quite a fragile implementation and currently requires us to maintain a population dataset in the Grapher codebase.

### PR DESCRIPTION
**Pull request by BhCh7051**
Fixes #1351 
"Hide countries with <1m people" checkbox and the corresponding Jest test removed.

### CODE REVIEW

**Comment by marcelgerber** [comment]
Thank you very much for the contribution @BhCh7051.

There is a lot of other code we can get rid of together with this checkbox, and I've added all of that as additional commits here.
Our svg tester is currently running on this code and looking for differences in rendered svgs. We will be waiting on its output before we can merge this PR, but that can hopefully happen sometime later today.

Again, thank you!

File: db/migration/1652693645426-RemoveMinPopulationFilter.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export class RemoveMinPopulationFilter1652693645426
+    implements MigrationInterface
+{
+    name = "RemoveMinPopulationFilter1652693645426"
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const tables = {
+            charts: "config",
+            suggested_chart_revisions: "suggestedConfig",
+        }
```
Comment (danielgavrilov) [inline db/migration/1652693645426-RemoveMinPopulationFilter.ts:12]: Nice that you included `suggested_chart_revisions`! I wonder if we should also include `chart_revisions`? I haven't done it consistently in the past, but it seems somewhat safe to do in this case

**Comment by danielgavrilov** [comment]
Looks like in most charts this was actually an improvement, but we are probably making worse:

- The COVID Explorer, which I think we can't do much about
- These 3 charts: http://localhost:3030/admin/test/embeds?ids=4704,4696,4695
   localhost_3030_admin_test_embeds_ids=4704,4696,4695 comparisonUrl=https%3A%2F%2Fourworldindata org

Should we maybe flag them to authors, in case they want to hide Turks and Caicos Island?

All else looks good! 🙌

**Comment by marcelgerber** [comment]
> Should we maybe flag them to authors, in case they want to hide Turks and Caicos Island?

Yes, will merge & deploy this now, and create an issue for the authors to have a look at these charts afterwards.

**Comment by danyx23** [comment]
@marcelgerber @danielgavrilov we should try in these cases to update the schema in these cases. One reason for this is that it is just a good idea and the other is that the bulk FASTT feature relies on the schema to create columns and patches. 

I'll create a follow up PR with the first schema migration and update the docs in https://github.com/owid/owid-grapher/tree/master/grapher/schema

**Comment by danielgavrilov** [comment]
@danyx23 Marcel updated the schema file, is there more we should be doing in cases like these?

**Comment by marcelgerber** [comment]
@danyx23 ah! I updated the schema file, but I assume that in this case I should've created a new version of the file, right? sorry about that.

I also just saw that the bulk FASTT editor _still_ shows `minPopulationFilter` as a column choice, even though I can't find any reference to it in the code any more. wondering where that is coming from.

**Comment by danyx23** [comment]
Ah sorry my bad, I didn't see the change in the schema file. All good! Destructive changes should lead to a new schema version but we are only using this internally so no problem for now

**Comment by danielgavrilov** [comment]
> I also just saw that the bulk FASTT editor _still_ shows `minPopulationFilter` as a column choice, even though I can't find any reference to it in the code any more. wondering where that is coming from.

The schema is fetched from a DigitalOcean bucket. We need to re-upload it to update it...

**Pull request merged**